### PR TITLE
[WIP] Create user and login with password

### PR DESCRIPTION
**Pull request by TimMikeladze**
My apologies for the giant PR. 

This PR introduces the core accounts package. It provides the client and server side logic for user creation and authentication.

### CODE REVIEW

File: src/client/module.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  isLoading: false,
+  user: null,
+  loggingIn: false,
+});
+
+const reducer = (state = initialState, action) => {
+  const nextState = state;
+  switch (action.type) {
+    case LOGIN: {
+      break;
```
Comment (grantga) [inline src/client/module.js]: Are these still to do on purpose?

Comment (TimMikeladze) [inline src/client/module.js]: Good call. The `LOGIN`case can be removed. `SET_USER` still needs to be filled out for reactivity in the ui when the user is logged in.

File: src/server/AccountsServer.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return this.instance.findUserByUsername(username, onlyId);
+  },
+  findUserById(userId: string): Promise<?UserObjectType> {
+    return this.instance.findUserById(userId);
+  },
+  addEmail(userId: string, newEmail: string, verified: boolean): Promise<void> {
+    return this.instance.addEmail(userId, newEmail, verified);
+  },
+  removeEmail(userId: string, newEmail: string): Promise<void> {
+    return this.instance.addEmail(userId, newEmail);
```
Comment (alexyuly) [inline src/server/AccountsServer.js]: Should this be `this.instance.removeEmail`?

File: src/server/AccountsServer.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  },
+  verifyEmail(userId: string, email: string): Promise<void> {
+    return this.instance.addEmail(userId, email);
```
Comment (alexyuly) [inline src/server/AccountsServer.js]: `this.instance.verifyEmail`?

File: src/server/AccountsServer.js (hunk 2 above)
Comment (TimMikeladze) [inline src/server/AccountsServer.js]: Eagle eye. Good catch. Will fix.

**Pull request merged**